Emulator/SPD.File.Emulator/Spd/SpdSpriteEntry.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Collections;
+using System.Diagnostics.CodeAnalysis;
+
+namespace SPD.File.Emulator.Spd;
+
+public struct SpdSpriteEntry
+{
+#pragma warning disable IDE0044 // Add readonly modifier
+#pragma warning disable IDE0051 // Remove unused private members
+    int _spriteId;
+    int _spriteTextureId;
+    int _unk08;
+    int _unk0c;
+    int _unk10;
+    int _unk14;
+    int _unk18;
+    int _unk1c;
+    int _spriteXPosition;
+    int _spriteYPosition;
+    int _spriteXLength;
+    int _spriteYLength;
+    int _unk30;
+    int _unk34;
+    int _spriteXScale;
+    int _spriteYScale;
+    int _unk40;
+    int _unk44;
+    int _unk48;
+    int _unk4c;
+    int _unk50;
+    int _unk54;
+    int _unk58;
+    int _unk5c;
+    int _unk60;
+    int _unk64;
+    int _unk68;
+    int _unk6c;
+    unsafe fixed byte _spriteName[48];
+
+    public readonly int GetSpriteId() => _spriteId;
+    public readonly int GetSpriteTextureId() => _spriteTextureId;
+    public SpdSpriteEntry SetTextureId(int id)
+    {
+        _spriteTextureId = id;
+        return this;
+    }
+
+#pragma warning restore IDE0044 // Add readonly modifier
+#pragma warning restore IDE0051 // Remove unused private members
+}
+
+public class SpdSpriteDictionary : IDictionary<int, SpdSpriteEntry>
+{
+    private readonly Dictionary<int, SpdSpriteEntry> SpriteDictionary;
+
+
+    public SpdSpriteDictionary()
+    {
+        SpriteDictionary = new Dictionary<int, SpdSpriteEntry>();
+    }
+
+    public SpdSpriteEntry this[int key]
+    {
+        get { return SpriteDictionary[key]; }
```

Comment:
This can be simplified to 
```
public SpdTextureEntry this[int key]
{
    get => _spriteDictionary[key];
    set => _spriteDictionary[key] = value;
}
```
And variable renamed to use underscore per C# style. 😉